shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java

```diff
@@ -636,17 +670,62 @@ public ResponseEntity<ByteArrayResource> downloadStatistics(
 		}
 
 		zipSingleFile(statisticsFile, zipFile);
-
-		byte[] data = Files.readAllBytes(zipFile.toPath());
-		ByteArrayResource resource = new ByteArrayResource(data);
-
 		statisticsFile.delete();
 
-		return ResponseEntity.ok()
-				.header(HttpHeaders.CONTENT_DISPOSITION, "attachment;filename=" + zipFile.getName())
-				.contentType(MediaType.MULTIPART_FORM_DATA)
-				.contentLength(data.length)
-				.body(resource);
+		event.setObjectId(String.valueOf(event.getId()));
+		event.setProgress(1f);
+		event.setMessage("Statistics fetched. Download available for 6 hours");
+		event.setStatus(ShanoirEvent.SUCCESS);
+		eventService.publishEvent(event);
 	}
 
+	@Override
+	public ResponseEntity<ByteArrayResource> downloadStatisticsByEventId(String eventId) throws IOException {
+		try {
+			String tmpDir = System.getProperty(JAVA_IO_TMPDIR);
+			File userDir = DatasetFileUtils.getUserImportDir(tmpDir);
+			File zipFile = new File(userDir + File.separator + "shanoirExportStatistics_" + eventId + ZIP);
+
+			byte[] data = Files.readAllBytes(zipFile.toPath());
+			ByteArrayResource resource = new ByteArrayResource(data);
+			deleteStats();
+
+			return ResponseEntity.ok()
+					.header(HttpHeaders.CONTENT_DISPOSITION, "attachment;filename=" + zipFile.getName())
+					.contentType(MediaType.MULTIPART_FORM_DATA)
+					.contentLength(data.length)
+					.body(resource);
+		} catch (Exception e) {
+			LOG.error("Error during download of statistics for event with id = " + eventId + ".");
+			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+		}
+	}
+
+	@Scheduled(cron = "0 0 6 * * *", zone="Europe/Paris")
+	public void deleteStats() {
+		try {
+			String tmpDir = System.getProperty(JAVA_IO_TMPDIR);
+			File userDir = DatasetFileUtils.getUserImportDir(tmpDir);
+			Path directoryPath = Paths.get(userDir.getPath());
+
+			long currentTime = System.currentTimeMillis();
+			long sixHoursInMillis = TimeUnit.HOURS.toMillis(6);
+			DirectoryStream<Path> directoryStream = Files.newDirectoryStream(directoryPath);
+
+			for (Path filePath : directoryStream) {
+				if (filePath.getFileName().toString().endsWith(".zip")) {
```

Comment:
Can you test if the file name starts with shanoirExportStatistics_  instead ?

Reply:
And maybe retrieve the event and update it so that the download button is not available anymore ?